File: solutions/punched-cards/punched-cards.js
"use strict";

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", (inputStdin) => {
	inputString += inputStdin;
});

process.stdin.on("end", (_) => {
	inputString = inputString
		.trim()
		.split("\n")
		.map((string) => {
			return string.trim();
		});

	main();
});

function readline() {
	return inputString[currentLine++];
}

// Make a Snippet for the code above this and then write your logic in main();

function main() {
	// Declare and read the number of test cases.
	var T;
	T = parseInt(readline());

	// Loop over the number of test cases.
	for (var test_no = 1; test_no <= T; test_no++) {
		process.stdout.write("Case #" + test_no + ": \n");
		solve();
	}
}

function solve() {
	// Declare variables N and M.
	var rows, cols;
	// Read the integers from the standard input.
	[rows, cols] = readline()
		.split(" ")
		.map((x) => parseInt(x));
	//get the real size
	rows = rows * 2 + 1;
	cols = cols * 2 + 1;

	//this will keep the state of the next character to be printed
	let nextIndexOdd = 0;
	let nextIndexEven = 0;

	//rotates the next character between + and -
	const nextOdd = () => {
		const chars = ["-", "+"];
		nextIndexOdd = (nextIndexOdd + 1) % chars.length;
		return chars[nextIndexOdd];
	};

	//rotates the next character between | and .
	const nextEven = () => {
		const chars = [".", "|"];
		nextIndexEven = (nextIndexEven + 1) % chars.length;
		return chars[nextIndexEven];
	};

	//loop through each row
	for (let row = 1; row <= rows; row++) {
		//give the correct characters to each row
		let next = row % 2 !== 0 ? nextOdd : nextEven;
		for (let col = 1; col <= cols; col++) {
			//the first square is always empty
			if (row <= 2 && col <= 2) {
				process.stdout.write(".");
			} else {
				process.stdout.write(next());
			}
		}
		//restart the character counter for the next line
		nextIndexEven = 0;
		nextIndexOdd = 0;
		process.stdout.write(`\n`);
	}
}

File: solutions/punched-cards/punchedCards.py
def punchCard(R, C):

    for r in range(R * 2 + 1):
        for c in range(C * 2 + 1):
            if r < 2 and c < 2:
                print(".", end="")
            else:
                if r % 2 == 0:
                    if c % 2 == 0:
                        print("+", end="")
                    else:
                        print("-", end="")
                else:
                    if c % 2 == 0:
                        print("|", end="")
                    else:
                        print(".", end="")
        print("")


# Read the number of test cases.
T = int(input())
# Loop over the number of test cases.
for test_no in range(1, T + 1):
    # get rows and columns
    (R, C) = tuple(map(int, input().split()))
    print("Case #%d:" % test_no)
    punchCard(R, C)
